edx_sga/static/js/src/edx_sga.js

```diff
@@ -1,22 +1,240 @@
 /* Javascript for StaffGradedAssignmentXBlock. */
 function StaffGradedAssignmentXBlock(runtime, element) {
+    function xblock($, _) {
+        var uploadUrl = runtime.handlerUrl(element, 'upload_assignment');
+        var downloadUrl = runtime.handlerUrl(element, 'download_assignment');
+        var annotatedUrl = runtime.handlerUrl(element, 'download_annotated');
+        var getStaffGradingUrl = runtime.handlerUrl(element, 'get_staff_grading_data');
+        var staffDownloadUrl = runtime.handlerUrl(element, 'staff_download');
+        var staffAnnotatedUrl = runtime.handlerUrl(element, 'staff_download_annotated');
+        var staffUploadUrl = runtime.handlerUrl(element, 'staff_upload_annotated');
+        var enterGradeUrl = runtime.handlerUrl(element, 'enter_grade');
+        var removeGradeUrl = runtime.handlerUrl(element, 'remove_grade');
+        var template = _.template($(element).find("#sga-tmpl").text());
+        var gradingTemplate;
 
-    function updateCount(result) {
-        $('.count', element).text(result.count);
-    }
+        function render(state) {
+            // Add download urls to template context
+            state.downloadUrl = downloadUrl;
+            state.annotatedUrl = annotatedUrl;
+            state.error = state.error ? state.error : false;
+
+            // Render template
+            var content = $(element).find("#sga-content").html(template(state));
+
+            // Set up file upload
+            $(content).find(".fileupload").fileupload({
+                url: uploadUrl,
+                add: function(e, data) {
+                    var do_upload = $(content).find(".upload").html('');
+                    $('<button/>')
+                        .text('Upload ' + data.files[0].name)
+                        .appendTo(do_upload)
+                        .click(function() {
+                            do_upload.text("Uploading...");
+                            data.submit();
+                        });
+                },
+                progressall: function(e, data) {
+                    var percent = parseInt(data.loaded / data.total * 100, 10);
+                    $(content).find(".upload").text(
+                        "Uploading... " + percent + "%");
```

Comment:
Just checked in with Sarina and this is not as hard of a requirement for our XBlocks, since we do not provide very clean solutions. I believe this XBlock is installed as a Django App so you may be able to use the i18n XBlock service, same as ORA2, but if that needs to be done separately from the initial installation that is fine.